test/functional/custom-transform.spec.ts

```diff
@@ -66,62 +125,51 @@ describe("custom transformation decorator", () => {
 
     });
 
-    it("versions and groups should work with @Transform decorator too", () => {
+    it("Non class objects should not break if transform is used", () => {
```

Comment:
I have no idea why the code in the routing controller calls logic like this. I just created the scenario from the routing library and gave you what you asked for. Either you can look at the scenario that causes the issue in the other lib or look at this one.